src/runners/cairoRunner.ts

```diff
@@ -112,7 +162,7 @@ export class CairoRunner {
   }
 
   getOutput() {
-    const builtins = this.program.builtins;
+    const builtins = (this.program as CairoZeroProgram).builtins;
```

Comment:
why is this cast needed?

Reply:
Cairo Zero artifacts have a global `builtins` field containing the 'output' builtin, whereas Cairo/Starknet artifacts have a `builtins` field nested in each function entrypoint which doesn't contain the 'output' builtin.

The output builtin is always used in `proof_mode` and for Starknet contracts. In the `cairo-vm` , for Cairo programs in ExecutionMode, it requires a flag (see `cairo1-run`), only inputs and return values of the type `Array<Felt252>` are expected/accepted.

The `getOutput()` method only has the logic for Cairo Zero artifacts. We could differenciate both artifacts and throw an exception for CairoProgram until the logic around input/return types is implemented.

The cast is needed to avoid a TypeScript error because there isn't a global `builtins` field in both artifacts.

CHANGELOG.md of the `cairo-vm`:
> [1.0.0-rc4] - 2024-07-05
>  * feat(BREAKING): Serialize inputs into output segment in cairo1-run crate:
>    * Checks that only `Array<Felt252>` can be received by the program main function when running with with either `--proof_mode` or `--append_return_values`.
>    * Copies the input value to the output segment right after the output in the format `[array_len, arr[0], arr[1],.., arr[n]]`.
>  * feat(BREAKING): Serialize `Array<Felt252>` return value into output segment in cairo1-run crate:
>    * Checks that only `PanicResult<Array<Felt252>>` or `Array<Felt252>` can be returned by the program when running with either `--proof_mode` or `--append_return_values`.
>  * Serializes return values into the output segment under the previous conditions following the format:
>    * `PanicResult<Array<Felt252>>` -> `[panic_flag, array_len, arr[0], arr[1],.., arr[n]]`
>    * `<Array<Felt252>` -> `[array_len, arr[0], arr[1],.., arr[n]]`
> 
> [1.0.0-rc2] - 2024-05-02
> * feat: Add flag to append return values to output segment when not running in proof_mode [#1646](https://github.com/lambdaclass/cairo-vm/pull/1646)
>  * Adds the flag `append_return_values` to both the CLI and `Cairo1RunConfig` struct.
>  * Enabling flag will add the output builtin and the necessary instructions to append the return values to the output builtin's memory segment.